src/stemmabench/algorithms/stemma.py

```diff
@@ -0,0 +1,209 @@
+                for lab in self.text_lookup:
+                    if value.text_lookup.get(lab) == None or not self.text_lookup[lab].__eq__(value.text_lookup[lab]):
+                        return False
+                return True
+        return False
+    
+    def __repr__(self) -> str:
+        """String representation of the tree."""
+        if self.fitted:
+            return "Tree(" + json.dumps(self.dict(), indent=2) + ")"
+        return "Empty"
+
+    def dump(self, folder: Union[str, None] = None) -> None:
+        """Dump the generated stemma into a folder:
+            - The texts in .txt file named after the manuscript label.
+            - The corresponding tree structure in edge file.
+            If folder is not specified will use the stemas path_to_folder attribute as path.
+
+        ### Args:
+            - folder (str, Optional): The folder where the text should be written.
+
+        ### Raises:
+            - ValueError: If folder is not specified and the folder_path variable is not set.
+        """
+        if not self.folder_path and not folder:
+            raise ValueError("This stemmas path_to_folder varaibles is not set. Therefor folder must be specified.")
+        if not folder and self.folder_path:
+            folder = self.folder_path
+        if not os.path.isdir(folder):
+                Path(folder).mkdir(exist_ok=True)
+        for key in self.text_lookup:
+            self.text_lookup[key].dump(folder)
+
+    def compute(self,
+                algo = None,
+                edge_file: Union[str, None] = None,
+                folder_path: Union[str, None] = None,
+                generation_info: Union[Dict[str, Any], None] = None,
+                *args,
+                **kargs) -> None:
+        """Builds the stemma based on given algorithm or edge file. Builds the stemma in place. 
+           Meaning it does not return anything.
+
+        ### Args:
+            - algo (StemmaAlgo, Optinal): indicating the function used to build the tree.
+            - edge_file (str, Optional): Path to the edge file used to build the tree. If given will ignore all other parameters.
+            - folder_path (str, Optional): The path to the folder containing all the stemma texts. 
+            If specified with the algo perameter this path will overwrite all other
+            - **kargs: The parameters to be passed to the algo parameter for the stemma generation.
+
+        ### Raises:
+            - ValueError: If edge_file is specified and path_to_folder is not specified.
+            - RuntimeError: If folder_path is not specified and both the _folder_path attribute and the algo folder_path attribute are set to None.
+            - RuntimeError: If both edge_file and algo pararmeters are unspecified.
+        """
+        self._generation_info = generation_info
+        if edge_file:
+            if not folder_path:
+                raise ValueError("If edge_file is specified path_to_folder must be specified.")
```

Comment:
Add f-string formatting in order to specify the missing file

Reply:
Error message removed because it warned about folder_path parameter that has since been removed